### PR TITLE
Fix js reader.js's skipGroup

### PR DESCRIPTION
**Pull request by BSBandme**
Previously skip group forgot to call skipField. 
Fix this and make the test case more complicated.

### CODE REVIEW

File: js/binary/reader.js [hunk 1]
```diff
@@ -389,27 +389,25 @@ jspb.BinaryReader.prototype.skipFixed64Field = function() {
  */
 jspb.BinaryReader.prototype.skipGroup = function() {
   // Keep a stack of start-group tags that must be matched by end-group tags.
```
Comment (TeBoring) [inline js/binary/reader.js]: Do you need to update the comment?

Comment (BSBandme) [reply js/binary/reader.js]: Done by deleting this comment.

File: js/binary/reader_test.js [hunk 2]
```diff
@@ -680,8 +680,10 @@ describe('binaryReaderTest', function() {
     var dummyMessage = /** @type {!jspb.BinaryMessage} */({});
     writer.writeGroup(5, dummyMessage, function() {
       writer.writeInt64(42, 42);
```
[the remark this answers is not in the document]
Comment (BSBandme) [reply js/binary/reader_test.js]: Previously the SkipGroup only consume the decoder by this.nextField, which only take Field number, so the case for "writer.writeInt64(42, 42)" has become one field with number 42 with type varint, and another field with number 42 / 8 = 5 with type 42 % 9 = 2 (length delimit) . And this don't affect reading next. 

But the new one will cause invalid varint with a random length delimit content.

And added another case which will cause unexpected end group for old code.

[the remark this answers is not in the document]
Comment (BSBandme) [reply js/binary/reader_test.js]: Done by changed the string to a bytes that will cause varint error and added a end group case in the nested group.

File: js/binary/reader_test.js [hunk 3]
```diff
@@ -681,10 +681,12 @@ describe('binaryReaderTest', function() {
...
       writer.writeInt64(44, 44);
-      writer.writeString(43, 'The quick brown fox jumps over the lazy dog');
+      writer.writeBytes(43, [255, 255, 255, 255, 255, 255, 255, 255, 255, 255]);
```
Comment (TeBoring) [inline js/binary/reader_test.js]: Please also make some comment for what this tests.

Comment (BSBandme) [reply js/binary/reader_test.js]: Done

File: js/binary/reader_test.js [hunk 4]
```diff
@@ -681,10 +681,12 @@ describe('binaryReaderTest', function() {
     writer.writeGroup(5, dummyMessage, function() {
       writer.writeInt64(42, 42);
       writer.writeInt64(44, 44);
```
[the remark this answers is not in the document]
Comment (BSBandme) [reply js/binary/reader_test.js]: Done.

[the remark this answers is not in the document]
Comment (TeBoring) [reply js/binary/reader_test.js]: Could you explain a little more about why 44 is an early end group.

File: js/binary/reader_test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (TeBoring) [reply js/binary/reader_test.js]: I think previously ```writer.writeString(43, "The quick brown fox jumps over the lazy dog")``` tests open group (field = 5) and end another group(field = 10). Could you still keep that test case? Perhaps you can make it easier by using ```writeInt64(43, 43)``` and ```writeInt64(84, 84)```?

File: js/binary/reader_test.js (hunk 3 above)
Comment (TeBoring) [reply js/binary/reader_test.js]: I am a little confused about how this is failed by an invalid varint. Does ```writeInt64(43, 15)``` has the same effect? (15=8 + 7, which creates a invalid tag 7)

Comment (BSBandme) [reply js/binary/reader_test.js]: Done.

File: js/binary/reader_test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (BSBandme) [reply js/binary/reader_test.js]: Done.

**Comment by xfxyjwf** [comment]
@BSBandme Yilun, can you send this as a google3 CL instead? None of us knows much about javascript and it's better to get someone from jspb team to look at it.

**Comment by BSBandme** [comment]
OK, I'll do that, although I think this is just a normal bug which is not JS related.

**Comment by BSBandme** [comment]
Is this OK to merge?

**Comment by xfxyjwf** [comment]
LGTM

**Pull request merged**